src/ProgramBuilder.ts

```diff
@@ -101,6 +101,8 @@ export class ProgramBuilder {
             this.options = util.normalizeAndResolveConfig(options);
             if (this.options.project) {
                 this.logger.log(`Using config file: "${this.options.project}"`);
+            } else if (this.options.noProject) {
+                this.logger.log(`No bsconfig.json file expected because using flag --noProject`);
```

Comment:
Let's move the `noProject` check first, so it will override the `.project` option.
```suggestion
            this.options = util.normalizeAndResolveConfig(options);
            if (this.options.noProject) {
                this.logger.log(`bsconfig.json loading disabled because 'noProject' option is set`);
            } else if (this.options.project) {
                this.logger.log(`Using config file: "${this.options.project}"`);
            } else
```